parser/start_tokens.coffee

```diff
@@ -12,7 +12,7 @@ class StartTokens
   @expression = [
     'INTLIT', 'FLOATLIT', 'true', 'false', # Primitive literals
     'ID', '->', 'trait', 'class',          # Misc other tokens
-    @iterable, tokens.reservedWords,
+    @iterable, tokens.beginnings,
```

Comment:
I'd rather have beginning only be specified in `start_tokens` and keep the reserved words in the scanner as they were before. I don't think logic for the parser should spill into any part of the scanner.